src/components/dataSource/DataSource.tsx

```diff
@@ -8,9 +8,11 @@ function DataSource() {
     const { panel } = useTheme();
     const [dataConnector, setDataConnector] = useState<ConnectorInstance | null>();
     const [firstRowInfo, setFirstRowInfo] = useState('');
```

Comment:
seems like it should be currentRow because you will have possibility to switch in follow up tickets